src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -38,6 +41,10 @@ public function has($relation, $operator = '>=', $count = 1, $boolean = 'and', ?
             }
 
             $relation = $this->getRelationWithoutConstraints($relation);
+        } else if ($relation instanceof Closure) {
+            $relation = Relation::noConstraints(function () use ($relation) {
+                return $relation($this->getModel());
+            });
         }
```

Comment:
Instead of essentially duplicating the `getRelationWithoutConstraints` method, why not update the method instead to handle strings or closures?

The `getRelationWithoutConstraints` method could look like this:

```php
    /**
     * Get the "has relation" base query instance.
     *
     * @param  (\Closure(TModel): \Illuminate\Database\Eloquent\Relations\Relation<*, *, *>)|string  $relation
     * @return \Illuminate\Database\Eloquent\Relations\Relation<*, *, *>
     */
    protected function getRelationWithoutConstraints($relation)
    {
        return Relation::noConstraints(function () use ($relation) {
            if (is_string($relation)) {
                return $this->getModel()->{$relation}();
            }

            return $relation($this->getModel());
        });
    }
```

and this code could be updated to:

```php
        if (! $relation instanceof Relation) {
            if (is_string($relation) && str_contains($relation, '.')) {
                return $this->hasNested($relation, $operator, $count, $boolean, $callback);
            }

            $relation = $this->getRelationWithoutConstraints($relation);
        } 
```